mealie/routes/spa/__init__.py

```diff
@@ -72,8 +72,24 @@ def inject_meta(contents: str, tags: list[MetaTag]) -> str:
 
 
 def inject_recipe_json(contents: str, schema: dict) -> str:
-    schema_as_html_tag = f"""<script type="application/ld+json">{json.dumps(jsonable_encoder(schema))}</script>"""
-    return contents.replace("</head>", schema_as_html_tag + "\n</head>", 1)
+    soup = BeautifulSoup(contents, "lxml")
+    schema_as_json = json.dumps(jsonable_encoder(schema))
+
+    script_tags = soup.find_all("script", {"type": "application/ld+json"})
+    for script_tag in script_tags:
+        try:
+            data = json.loads(script_tag.string)
+            if data.get("@type") == "Recipe":
+                script_tag.string = schema_as_json
+                break
+        except json.JSONDecodeError:
+            continue
+    else:
+        schema_as_html_tag = soup.new_tag("script", type="application/ld+json")
+        schema_as_html_tag.string = schema_as_json
+        soup.head.append(schema_as_html_tag)
+
+    return str(soup)
```

Comment:
Why do we need this?

Reply:
I added this just to ensure we have exactly one set of metadata (i.e. "replace the existing metadata or add new metadata"). I don't know if it's necessary since the global state issue is fixed, but the redundancy felt nice in case something else adds metadata for some reason.

I can remove it though, pretty sure this PR works without it

Reply:
I think either or would be good. Either we can re-use the same string reference and modify it, or make a copy every request. No sense it spinning up Beautiful Soup on every request if we don't have to.